### DISCUSSION TITLE
How to setup external dev server

### DISCUSSION THREAD

**Discussion by Eskaan** [wailsapp/wails#1036] [Q&A]
Hello, I have a few question about the "devserverurl" property (I hope this is the right place to ask).
1) Why does the project still compile the frontend when setting the variable? (Because there should be a server serving those files)
2) Why do you still start the DevWebServer (And I think also use it) when I set this - Also the contents should be delivered by the external server (For example set up in frontend:dev).

These two questions result of my efforts of binding an external development server (`npm run dev` for a svelte project) to the wails app. I think I don't really understand something or just didn't get it here. 

Also I got this error: 
```
Error during retrieving assetdir: Get "http://localhost:34115/wails/assetdir": dial tcp 127.0.0.1:34115: connect: connection refused
Reloading couldn't be triggered: Please specify -assetdir or -reloaddirs
```
Whilst having the wails.json devserverurl set to `http://localhost:3000` which is pretty confusing to me. I hope someone is able to help me. Thanks for your efforts :)

**Comment by leaanthony** [comment]
Thanks for the detailed report! Firstly, that option is kinda experimental. The idea is that all asset requests are directed to the given URL. The last report back was that it was working ok so if you can set up a test project that reproduces the issue then perhaps we can look at fixing it?

**Reply by Eskaan** [reply]
Will do that, thanks for the fast reply and until after Christmas (Merry Christmas to everyone reading this)

**Comment by Eskaan** [comment]
(I am a bit angry my long and detailed report just got lost by me clicking a link)
So, we over at [GDLauncher](https://github.com/gorilla-devs/GDLauncher/) are planing to [port our app](https://github.com/gorilla-devs/GDLauncher/tree/gdlib-refactor/reindex) over to a backend-wails frontend-svelte combination and we would very appreciate a working reload feature, even tho its not necessary because we can also just develop directly in a web browser.

I am on a MacBook Air 2015 whit MacOS 12.0.1 (Monterey),
Go version is 1.17.5 (just installed the default because I thought it could be caused by my different installation by Brew)
EDIT: Oh yeah and also you can find the test repo on https://github.com/Code-Ac/WailsSvelteBase

So, my usual setup is that I run `cd frontend; npm i; npm run dev` and then switch to another task/wondow/tab to run `wails dev` in it. This results in:
```
Wails CLI v2.0.0-beta.23

Executing: go mod tidy
Executing: wails generate module
Building application for development...
Installing frontend dependencies: Done.
Compiling frontend: Done.
Compiling application: Done.
Packaging application: Done.

Watching (sub)/directory: /Users/kinder/Documents/GitHub/wails-svelte-base
Using Dev Server URL: http://localhost:3000
Using reload debounce setting of 100 milliseconds
INF | Serving assets from disk: /Users/kinder/Documents/GitHub/wails-svelte-base
Added new directory to watcher: .
Added new directory to watcher: /Users/kinder/Documents/GitHub/wails-svelte-base/frontend/src/wailsjs/go
DEB | [DesktopAssetServer] Loading file: _app/assets/start-61d1577b.css
DEB | [DesktopAssetServer] Loading file: _app/start-3968025c.js
DEB | [DesktopAssetServer] Loading file: _app/chunks/vendor-356027b4.js
DEB | [DesktopAssetServer] Loading file: _app/assets/pages/__layout.svelte-a7861b78.css
DEB | [DesktopAssetServer] Loading file: _app/assets/pages/index.svelte-c63fe1c6.css
DEB | [DesktopAssetServer] Loading file: _app/error.svelte-3d1b715b.js
DEB | [DesktopAssetServer] Loading file: _app/pages/__layout.svelte-dae11831.js
DEB | [DesktopAssetServer] Loading file: _app/pages/index.svelte-abf06611.js
DEB | [DesktopAssetServer] Loading file: _app/assets/svelte-logo-87df40b8.svg
DEB | [DesktopAssetServer] Loading file: svelte-welcome.png
DEB | [DesktopAssetServer] Loading file: svelte-welcome.webp
```
Which is fine for me even tho it still compiles the code and then serves then initial site by this compiled code. But if I then try editing for example /routes/index.svelte and save:
```
Error during retrieving assetdir: Get "http://localhost:34115/wails/assetdir": dial tcp 127.0.0.1:34115: connect: connection refused
Reloading couldn't be triggered: Please specify -assetdir or -reloaddirs
```
I think this is caused by wails trying to serve files from the not running default configuration whilst ignoring the set devserverurl property. The part of the added `wails/assetdir` directory to the url is also interesting but could also result from a wrongly set assetdir variable (I think). Main problem here is that it serves from the wrong server.

I also posted some of the features I would like to see integrated in #999 but no pressure here, they are mostly aesthetic things.

Thanks for reading (and I hope soon fixing). I think you can convert this to an issue at this point.
